### PR TITLE
tests: benchmarks: multicore: idle: Add GPIO wake-up manner

### PR DESCRIPTION
**Pull request by adamkondraciuk**
Extend the sample by adding configuration that allows wake-up using sw0 button.

### CODE REVIEW

File: tests/benchmarks/multicore/idle_gpio/boards/nrf54h20dk_nrf54h20_cpuapp_s2ram.overlay [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                        compatible = "zephyr,power-state";
+                        power-state-name = "suspend-to-idle";
+                        min-residency-us = <100000>;
+                };
+
+                s2ram: s2ram {
+                        compatible = "zephyr,power-state";
+                        power-state-name = "suspend-to-ram";
+                        min-residency-us = <800000>;
+                        /* TODO: replace with a busy loop in S2RAM entering procedure; replace 1000 with real value here */
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_gpio/boards/nrf54h20dk_nrf54h20_cpuapp_s2ram.overlay:19]: obsolete

File: tests/benchmarks/multicore/idle_gpio/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      remote_CONFIG_UART_CONSOLE=n remote_CONFIG_SERIAL=n remote_CONFIG_GPIO=y
+      remote_CONFIG_BOOT_BANNER=n remote_CONFIG_NRFS_MRAM_SERVICE_ENABLED=n
+      remote_CONFIG_CLOCK_CONTROL=n
+      DTC_OVERLAY_FILE="boards/nrf54h20dk_nrf54h20_cpuapp_s2ram.overlay"
+      remote_DTC_OVERLAY_FILE="boards/nrf54h20dk_nrf54h20_cpurad_s2ram.overlay"
+    harness: pytest
+    harness_config:
+      fixture: ppk_power_measure
+      pytest_root:
+        - "${CUSTOM_ROOT_TEST_DIR}/test_measure_power_consumption.py::test_measure_and_data_dump_power_consumption"
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_gpio/testcase.yaml:33]: this will not work for gpio wakeup, it is prepared only for periodic wakeups.
For now that part needs to be skipped.

File: tests/benchmarks/multicore/idle_gpio/src/main.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	}
+	LOG_INF("Multicore idle_gpio test on %s", CONFIG_BOARD_TARGET);
+	while (1) {
+		LOG_INF("Multicore idle_gpio test iteration %u", cnt++);
+#if IS_ENABLED(CONFIG_SOC_NRF54H20_CPUAPP)
+		pm_state_set(PM_STATE_SUSPEND_TO_RAM, 0);
+#elif IS_ENABLED(CONFIG_SOC_NRF54H20_CPURAD)
+		pm_state_set(PM_STATE_RUNTIME_IDLE, 0);
+#endif
+		k_busy_wait(1000000);
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_gpio/src/main.c:47]: what about k_sleep and idle state?

Comment (adamkondraciuk) [reply tests/benchmarks/multicore/idle_gpio/src/main.c:47]: if you call k_sleep here, the gpio will wake up the CPU, but the scheduler will force sleeping again. We are introducing sleep state by  pm_state_set() while k_busy_wait() is simulating 1s of active state of the CPU

**Review by nordic-piks — APPROVED** [review]

File: tests/benchmarks/multicore/idle_gpio/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_gpio/CMakeLists.txt:2]: If the file is new, then year shall be 2024. Issue is in other files too.

File: tests/benchmarks/multicore/idle_gpio/Kconfig
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_gpio/Kconfig:1]: Is this file needed? It doesn't contain KConfig definition.

File: tests/benchmarks/multicore/idle_gpio/README.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+Testing
+=======
+
+After programming the test to your development kit, complete the following steps to test it:
+
+#. Connect the PPK2 Power Profiler Kit or other current measurement device.
+#. Reset the kit.
+#. You will observe low current consumption as all cores are suspended.
+#. When the button is pressed callback is user executed.
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_gpio/README.rst:62]: "callback is user executed" - looks like "user" shall be removed.

**Review by nordic-segl — APPROVED** [review]

**Review by nordic-segl — APPROVED** [review]

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: tests/benchmarks/multicore/idle_gpio/README.rst [hunk 6]
```diff
@@ -0,0 +1,66 @@
+.. _multicore_idle_gpio_test:
+
+Multicore idle GPIO test
+########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The test benchmarks the idle behavior of an application that runs on multiple cores. It uses a pin as a wake-up source.  It uses a  as a wake-up source.
```
Comment (wiba-nordic) [inline tests/benchmarks/multicore/idle_gpio/README.rst:10]:
```suggestion
The test benchmarks the idle behavior of an application that runs on multiple cores.
It uses a pin as a wake-up source.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: tests/benchmarks/multicore/idle/README.rst [hunk 7]
```diff
@@ -7,7 +7,7 @@ Multicore idle test
    :local:
    :depth: 2
 
-The test benchmarks the idle behavior of an application that runs on multiple cores.
+The test benchmarks the idle behavior of an application that runs on multiple cores. It uses a system timer as a wake-up source.
```
Comment (wiba-nordic) [inline tests/benchmarks/multicore/idle/README.rst:10]:
```suggestion
The test benchmarks the idle behavior of an application that runs on multiple cores.
It uses a system timer as a wake-up source.
```

**Review by nordic-segl — APPROVED** [review]
